state/head-mpt-state-network.md

```diff
@@ -0,0 +1,267 @@
+# Execution Head-MPT State Network
+
+| 🚧 THE SPEC IS IN A STATE OF FLUX AND SHOULD BE CONSIDERED UNSTABLE 🚧 <br> _Clients should implement Account Trie first and reevaluate poposed approach for Storage Tries_ |
+|-|
+
+This document is the specification for the Portal Network that supports on-demand availability of
+close to the head of the chain, Merkle Patricia Trie State data from the execution chain.
+
+While similar to the [Execution State network](state-network.md), it has some unique features:
+
+- It supports direct lookup of the account state, contract's code and contract's storage
+- It only stores state that is present in any of the latest 256 blocks
+- Nodes are responsible for storing subtree of the entire state trie
+
+## Overview
+
+The Execution Head-MPT State Network is a
+[Kademlia](https://pdos.csail.mit.edu/~petar/papers/maymounkov-kademlia-lncs.pdf) DHT that uses the
+[Portal Wire Protocol](../portal-wire-protocol.md) to establish an overlay network on top of the
+[Discovery v5](https://github.com/ethereum/devp2p/blob/master/discv5/discv5-wire.md) protocol.
+
+Nodes are responsible for storing fixed state subtree, across all 256 recent blocks.
+
+Nodes are expected to have access to the latest 256 block headers, which they will use to validate
+content and handle re-orgs. The [History](../history/history-network.md) and
+[Beacon](../beacon-chain/beacon-network.md) networks can be used for this purpose, but
+implementations can use other out-of-protocol solutions as well.
+
+Content is gossiped as block's trie-diff subtries. This provides very efficient way to keep node's
+subtrie updated as the chain progresses.
+
+### Data
+
+The network stores execution layer state content, which encompases the following data from the
+latest 256 blocks:
+
+- Block trie-diffs
```

Comment:
Would like to get validation of this concept done since it's a core building block of the network designs.  We need test data for:

- A block level diff that is correct (minimal and complete)
- A block level diff that contains extra data and the previous block level proof that allows detection of this.
- A block level diff that omits a change to the state and the previous block level proof that allows detection of this.